grc/utils/reference_number.py

```diff
@@ -1,23 +1,4 @@
-import random
-from grc.models import db, Application
-
-
-def reference_number_generator(email):
-    unambiguous_letters_and_numbers = "ABCDEFGHJKMNPQRSTUVWXYZ23456789"
-    ref_number = ''.join(random.choices(unambiguous_letters_and_numbers, k=8))
-    application_record = Application.query.filter_by(reference_number=ref_number).first()
-
-    if application_record is None:
-        try:
-            record = Application(reference_number=ref_number, email=email)
-            db.session.add(record)
-            db.session.commit()
-            return ref_number
-        except ValueError:
-            print("Oops!  Something went wrong.")
-    else:
-        print("Reference number exists, trying again")
-        reference_number_generator(email)
```

Comment:
This has moved to `data_store.py`
https://github.com/cabinetoffice/grc-app/blob/master/grc/business_logic/data_store.py#L72-L79